From db06590d50db1295891a6f9277a162d229049ffe Mon Sep 17 00:00:00 2001
From: Nicolas Boichat <drinkcat@google.com>
Date: Mon, 30 May 2016 14:40:52 +0800
Subject: [PATCH] FIXUP: egl/android: Add support for YV12 pixel format

Correctly setup the formats: YV12 is YVU420, not YUV420.

BUG=chrome-os-partner:53652
TEST=Boot oak-cheets, play Youtube video

Change-Id: Iadad3a658eb75e3c70936d7ac0df6ae1a3ca3a33
Signed-off-by: Nicolas Boichat <drinkcat@google.com>
---
 src/egl/drivers/dri2/platform_android.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/egl/drivers/dri2/platform_android.c b/src/egl/drivers/dri2/platform_android.c
index 2d2790d..7e74433 100644
--- a/src/egl/drivers/dri2/platform_android.c
+++ b/src/egl/drivers/dri2/platform_android.c
@@ -87,7 +87,7 @@ static unsigned int get_fourcc_format(int hal_format)
       format = __DRI_IMAGE_FOURCC_XBGR8888;
       break;
    case HAL_PIXEL_FORMAT_YV12:
-      format = __DRI_IMAGE_FOURCC_YUV420;
+      format = __DRI_IMAGE_FOURCC_YVU420;
       break;
    default:
       break;
@@ -431,11 +431,11 @@ dri2_create_image_android_native_buffer(_EGLDriver *drv, _EGLDisplay *disp,
    case HAL_PIXEL_FORMAT_YV12:
       /* Y plane is at offset 0 and fds[0] is already set. */
       /* Cr plane is located after Y plane */
-      offsets[2] = offsets[0] + strides[0] * buf->height;
-      strides[2] = ALIGN(strides[0] / 2, 16);
+      offsets[1] = offsets[0] + strides[0] * buf->height;
+      strides[1] = ALIGN(strides[0] / 2, 16);
       /* Cb plane is located after Cr plane */
-      offsets[1] = offsets[2] + strides[2] * buf->height / 2;
-      strides[1] = strides[2];
+      offsets[2] = offsets[1] + strides[1] * buf->height / 2;
+      strides[2] = strides[1];
       total_planes = 3;
       break;
    default:
-- 
2.8.0.rc3.226.g39d4020

